### PR TITLE
Removes PhysicalFS create_root calls

### LINKED ISSUE #677: Empty resources directory is created when creating Context.
**Describe the bug**
When creating a `Context` using `ContextBuilder::build` an empty resource directory is created next to the application.

**To Reproduce**
Create an application with the following code and run it:
```rust
pub fn main() -> ggez::GameResult {
    let cb = ggez::ContextBuilder::new("resources-test", "Eran Cohen").window_setup(
        ggez::conf::WindowSetup::default()
    );
    let (ctx, event_loop) = &mut cb.build()?;
}
```
**Expected behavior**
Calling `ContextBuilder::build` should **not** create an empty resource directory.

**The source of the behavior**
By default, `ContextBuilder::load_conf_file` is set to `true`, so when `build` is called the following line in *context.rs* is run: (line 299)
```rust
fs.read_config().unwrap_or(self.conf)
```
`read_config` calls `is_file` (*filesystem.rs*, line 327) which then calls `vfs.metadata`(*filesystem.rs*, line 233) which then calls another `vfs.metadata` (*vfs.rs*, line 499) and brings us to *vfs.rs*, line 350: 

```rust
fn metadata(&self, path: &Path) -> GameResult<Box<dyn VMetadata>> {
    self.create_root()?;
    let p = self.to_absolute(path)?;
    p.metadata()
        .map(|m| Box::new(PhysicalMetadata(m)) as Box<dyn VMetadata>)
        .map_err(GameError::from)
}
```
which as can be seen, calls `create_root` and creates the directory.

**Hardware and Software:**
 - ggez version: *0.5.1*
 - OS: *Windows 10*, *Windows 7*
 - Graphics card: *Intel HD Graphics 530*
 - Graphics card drivers: *Intel*

**Workaround**
Creating a `Context` without trying to load from a configuration file doesn't create the directory:
```rust
let cb = ggez::ContextBuilder::new("life-rs", "Eran Cohen").with_conf_file(false);
```

### PR DESCRIPTION
**Pull request by nobbele**
I don't see why it even needs to do that.
Fixes #677

### CODE REVIEW

**Comment by icefoxen** [comment]
It does it so it never needs to check whether the various config/save/etc directories exist, they just always do.  Whether it *still* needs to do this, idk.

**Comment by PSteinhaus** [comment]
> Whether it _still_ needs to do this, idk.

It does. I just tested it and the logging and files examples fail using this PR. The astroblasto example also throws an error when trying to create a screenshot.

I guess we need these checks.
But maybe we can find out why the empty directory is created even when there is no need for it.
Going through the code I don't see why ```create_root``` gets called at all when we build the context... maybe I'll try to debug it with gdb tomorrow.

File: src/vfs.rs [hunk 1]
```diff
@@ -275,7 +262,7 @@ impl VFS for PhysicalFS {
             );
             return Err(GameError::FilesystemError(msg));
         }
-        self.create_root()?;
+
```
Comment (PSteinhaus) [inline src/vfs.rs:265]:
```suggestion
        if open_options.create {
            self.create_root()?;
        }
```

File: src/vfs.rs [hunk 2]
```diff
@@ -293,7 +280,7 @@ impl VFS for PhysicalFS {
                     .to_string(),
             ));
         }
-        self.create_root()?;
+
```
Comment (PSteinhaus) [inline src/vfs.rs:283]:
```suggestion
        self.create_root()?;
```

File: src/vfs.rs [hunk 3]
```diff
@@ -239,19 +239,6 @@ impl PhysicalFS {
...
-    /// This way we can not create the directory until it's
-    /// actually used, though it IS a tiny bit of a performance
-    /// malus.
-    fn create_root(&self) -> GameResult {
-        if !self.root.exists() {
-            fs::create_dir_all(&self.root).map_err(GameError::from)
-        } else {
-            Ok(())
-        }
-    }
```
Comment (PSteinhaus) [inline src/vfs.rs:254]: If we want to use this in ```mkdir``` and ```open_options``` we need to keep it.

**Review by PSteinhaus — COMMENTED** [review]
I just tried all the relevant examples with these changes and they seem to work fine now.

**Comment by PSteinhaus** [comment]
I think this is ready to merge.

**Review by IGBC — APPROVED** [review]

**Pull request merged**